FEXCore/Source/Interface/IR/Passes/ConstProp.cpp

```diff
@@ -940,6 +705,8 @@ void ConstProp::ConstantInlining(IREmitter* IREmit, const IRListView& CurrentIR)
 
       uint64_t Constant2 {};
       if (Op->OffsetType == MEM_OFFSET_SXTX && IREmit->IsValueConstant(Op->Offset, &Constant2)) {
+        Constant2 *= Op->OffsetScale;
```

Comment:
Is this correct? Wouldn't the Operation still have OffsetScale set to its scale and now the constant has been scaled? So if the backend starts caring about scale+imm, the logic is now disjoint and odd from the original representation.

Reply:
Oh, yeah, good point - ConstProp should set OffsetScale = 1. Will fix.

Reply:
rewrote this commit